base/sim/FairMCApplication.cxx

```diff
@@ -1338,4 +1291,29 @@ void FairMCApplication::UndoGeometryModifications()
     gGeoManager->ClearPhysicalNodes(kFALSE);
 }
 
-ClassImp(FairMCApplication);
+void FairMCApplication::ConstructSensitiveDetectors()
+{
+    std::map<std::string, FairModule*> cloneVolumeMap;
+
+    for (auto const& x : fMapSensitiveDetectors) {
+        std::string volName = x.first;   //.substr(0, x.first.find("#", 0));
```

Comment:
Could you try a string_view here?

```suggestion
        std::string_view volName = x.first;   //.substr(0, x.first.find("#", 0));
```

Reply:
Ain't that easy. Should I change the whole map? 
```
FairRoot/base/sim/FairMCApplication.cxx:1301:38: error: no matching member function for call to 'find'
            auto it = cloneVolumeMap.find(volName);
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/c++/v1/map:1405:14: note: candidate function not viable: no known conversion from 'std::string_view' (aka 'basic_string_view<char>') to 'const std::map<std::string, FairModule *>::key_type' (aka 'const std::string') for 1st argument
    iterator find(const key_type& __k)             {return __tree_.find(__k);}
```

Reply:
No, leave the map as is. I was just hoping for the string_view to be a little more efficient. But it needs to be converted to a std::string for those errors anyway.

Thanks for trying it!